File: week2/Erenimo3442/valid_parantheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for i in s:
            if i == "(" or i == "[" or   i == "{":
                stack.append(i)
            elif i == ")":
                if not stack or not stack[-1] == "(":
                    return False
                    break
                else:
                    stack.pop(-1)
            elif i == "]":
                if not stack or not stack[-1] == "[":
                    return False
                    break
                else:
                    stack.pop(-1)
            elif i == "}":
                if not stack or not stack[-1] == "{":
                    return False
                    break
                else:
                    stack.pop(-1)
        return not stack

File: week2/AhmetTahaSener/ValidParantheses.cs
public class Solution
{
    public bool IsValid(string s)
    {
        Stack<char> character = new Stack<char>();

        for (int i = 0; i < s.Length; i++)
        {
            character.Push(s[i]);
            IsCompleted(character);
        }
        return (character.Count == 0);
    }

    public void IsCompleted(Stack<char> chars)
    {
        if (chars == null || chars.Count == 1)
        {
            return;
        }
        char cClose = chars.Pop();
        char cOpen;
        int top = chars.Count - 1;
        switch (cClose)
        {
            case ')':
                cOpen = chars.Pop();
                if (cOpen == '(')
                {
                    break;
                }
                chars.Push(cOpen);
                chars.Push(cClose);
                break;

            case '}':
                cOpen = chars.Pop();
                if (cOpen == '{')
                {
                    break;
                }
                chars.Push(cOpen);
                chars.Push(cClose);
                break;
            case ']':
                cOpen = chars.Pop();
                if (cOpen == '[')
                {
                    break;
                }
                chars.Push(cOpen);
                chars.Push(cClose);
                break;
            default:
                chars.Push(cClose);
                break;
        }
    }
}
